test/Microsoft.Azure.Devices.Edge.Test.Common/LeafDevice.cs

```diff
@@ -386,9 +388,16 @@ static Task<MethodResponse> DirectMethod(MethodRequest request, object context)
             return Task.FromResult(new MethodResponse(request.Data, (int)HttpStatusCode.OK));
         }
 
-        class FailingTLSConnectionErrorDetectionStrategy : ITransientErrorDetectionStrategy
+        // This error detection strategy is intended for SDK clients connecting
+        // to EdgeHub encountering a variety of issues.
+        //
+        // Can be removed when the below are fixed:
+        // 1 (AuthenticationException) - Sometimes tls auth error occurs because EdgeHub sends an unexpected message (work item 14057676).
+        // 2 (ObjectDisposedException) - Devices SDK Issue: ObjectDisposed exception (https://github.com/Azure/azure-iot-sdk-csharp/issues/2337)
+        // 3 (InvalidOperationException) - Devices SDK Issue: No authenticated context (https://github.com/Azure/azure-iot-sdk-csharp/issues/2353)
+        class FailingConnectionErrorDetectionStrategy : ITransientErrorDetectionStrategy
         {
-            public bool IsTransient(Exception ex) => ex is ObjectDisposedException || ex is AuthenticationException;
+            public bool IsTransient(Exception ex) => ex is ObjectDisposedException || ex is AuthenticationException || ex is InvalidOperationException;
```

Comment:
InvalidOperationException is a very Generic Exception to call Transient and may hide some other issues. Can we add another level of granularity something like ex.message.contains("...")?